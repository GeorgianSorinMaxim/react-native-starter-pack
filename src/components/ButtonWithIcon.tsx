import * as React from "react";
import { StyleSheet, Text, View } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import { RectButton } from "react-native-gesture-handler";

import Colors from "../constants/Colors";

type ButtonWithIconProps = {
  label: string;
  icon: string;
  onPress: () => void;
};

const ButtonWithIcon = ({ label, icon, onPress }: ButtonWithIconProps) => {
  return (
    <RectButton style={styles.option} onPress={onPress}>
      <View style={styles.buttonContainer}>
        <View style={styles.iconContainer}>
          <Icon name={icon} size={18} color={Colors.gold} />
        </View>
        <Text style={styles.optionText}>{label}</Text>
      </View>
    </RectButton>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: "row",
  },
  iconContainer: {
    marginRight: 12,
  },
  option: {
    paddingVertical: 15,
    paddingHorizontal: 15,
    borderColor: Colors.grey,
    borderWidth: StyleSheet.hairlineWidth,
    marginTop: -1,
    marginBottom: 1,
  },
  optionText: {
    marginTop: 1,
    fontSize: 12,
    letterSpacing: 4,
    alignSelf: "flex-start",
    textTransform: "uppercase",
  },
});

export default ButtonWithIcon;
