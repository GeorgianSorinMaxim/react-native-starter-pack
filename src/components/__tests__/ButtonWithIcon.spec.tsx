import * as React from "react";
import renderer from "react-test-renderer";

import ButtonWithIcon from "../ButtonWithIcon";

it(`renders correctly`, () => {
  const defaultProps = {
    label: "Google",
    onPress: () => {},
    isLastOption: false,
  };
  const tree = renderer.create(<ButtonWithIcon {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
