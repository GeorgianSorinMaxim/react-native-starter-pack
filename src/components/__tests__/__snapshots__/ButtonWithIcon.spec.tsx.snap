// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "borderColor": "#BABABA",
        "borderWidth": 0.5,
        "marginBottom": 1,
        "marginTop": -1,
        "paddingHorizontal": 15,
        "paddingVertical": 15,
      },
    ]
  }
  underlayColor="black"
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#D4AF37",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ?
      </Text>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 12,
          "letterSpacing": 4,
          "marginTop": 1,
          "textTransform": "uppercase",
        }
      }
    >
      Google
    </Text>
  </View>
</RNGestureHandlerButton>
`;
